internal/jobservice/repository/sql_job_service.go

```diff
@@ -0,0 +1,266 @@
+	jobLock       sync.RWMutex
+	subscribeMap  map[string]*SubscribeTable
+	subscribeLock sync.RWMutex
+}
+
+func NewJobStatus(jobMap map[string]*JobTable, subscribeMap map[string]*SubscribeTable) *JobStatus {
+	return &JobStatus{jobMap: jobMap, subscribeMap: subscribeMap}
+}
+
+// SQLJobService for persisting to DB.
+type SQLJobService struct {
+	jobStatus        *JobStatus
+	jobServiceConfig *configuration.JobServiceConfiguration
+	db               *sql.DB
+}
+
+func NewSQLJobService(jobMap *JobStatus, config *configuration.JobServiceConfiguration, db *sql.DB) *SQLJobService {
+	return &SQLJobService{jobStatus: jobMap, jobServiceConfig: config, db: db}
+}
+
+type SubscribedTuple struct {
+	Queue  string
+	JobSet string
+}
+
+// Create a Table from a hard-coded schema.
+// Open to suggestions on how to make this better
+func (s *SQLJobService) CreateTable() {
+	_, err := s.db.Exec("DROP TABLE IF EXISTS jobservice")
+	if err != nil {
+		panic(err)
+	}
+	_, err = s.db.Exec(`
+CREATE TABLE jobservice (
+Queue TEXT,
+JobSetId TEXT,
+JobId TEXT,
+JobResponseState TEXT,
+JobResponseError TEXT,
+Timestamp INT
+)`)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// Get the JobStatus given the jodId
+func (s *SQLJobService) GetJobStatus(jobId string) (*js.JobServiceResponse, error) {
+	s.jobStatus.jobLock.RLock()
+	jobResponse, ok := s.jobStatus.jobMap[jobId]
+	s.jobStatus.jobLock.RUnlock()
+	if !ok {
+		sqlResponse, errSql := s.GetJobStatusSQL(jobId)
+		if sqlResponse != nil {
+			return sqlResponse, errSql
+		}
+	}
+
+	return &jobResponse.jobResponse, nil
+}
+
+// If our in memory map no longer contains the status, we should query the row to get it
+// If that doesn't exist then we assume the job doesn't exist yet.
+func (s *SQLJobService) GetJobStatusSQL(jobId string) (*js.JobServiceResponse, error) {
+	row := s.db.QueryRow("SELECT JobResponseState, JobResponseError FROM jobservice WHERE JobId=?", jobId)
+	var jobState string
+	var jobError string
+	err := row.Scan(&jobState, &jobError)
+	if err == sql.ErrNoRows {
+		return &js.JobServiceResponse{State: js.JobServiceResponse_JOB_ID_NOT_FOUND}, nil
+	} else if err != nil {
+		return nil, err
+	}
+	jobProtoResponse := &js.JobServiceResponse{Error: jobError}
+	switch jobState {
+	case "SUBMITTED":
+		jobProtoResponse.State = js.JobServiceResponse_SUBMITTED
+	case "DUPLICATE_FOUND":
+		jobProtoResponse.State = js.JobServiceResponse_DUPLICATE_FOUND
+	case "RUNNING":
+		jobProtoResponse.State = js.JobServiceResponse_RUNNING
+	case "FAILED":
+		jobProtoResponse.State = js.JobServiceResponse_FAILED
+	case "SUCCEEDED":
+		jobProtoResponse.State = js.JobServiceResponse_SUCCEEDED
+	case "CANCELLED":
+		jobProtoResponse.State = js.JobServiceResponse_CANCELLED
+	case "JOB_ID_NOT_FOUND":
+		jobProtoResponse.State = js.JobServiceResponse_JOB_ID_NOT_FOUND
+	}
+	return jobProtoResponse, nil
+}
+
+// Update in memory JobStatus Map with jobId and our JobTable
+func (s *SQLJobService) UpdateJobServiceDb(jobTable *JobTable) {
```

Comment:
I would change the name of this function as it's a bit weird to have a function called `UpdateJobServiceDb` that doesn't actually update the DB.

Reply:
Function actually does update a database now.